web_search_with_and/static/src/js/search.js

```diff
@@ -0,0 +1,55 @@
+odoo.define('web_search_with_and', function (require) {
+    "use strict";
+
+    var SearchView = require('web.SearchView');
+    var Backbone = window.Backbone;
+
+    SearchView.include({
+        // Override the base method to detect a 'shift' event
+        select_completion: function (e, ui) {
+            if (e.shiftKey
+                && ui.item.facet.values
+                && ui.item.facet.values.length
+                && String(ui.item.facet.values[0].value).trim() !== "") {
+                // In case of an 'AND' search a new facet is added regarding of the previous facets
+                e.preventDefault();
+
+                this.query.add(ui.item.facet, {shiftKey: true});
+            } else {
+
+                return this._super.apply(this, arguments);
+            }
+        }
+    });
+
+    SearchView.SearchQuery.prototype = SearchView.SearchQuery.extend({
+        // Override the odoo method to (conditionally) add a search facet even if a existing
```

Comment:
can't we use `include` + `_super` here as well?

Reply:
@simahawk where I can find documentation about include and _super?

Reply:
a couple of lines above? :smile: 
I'm just guessing. Maybe the original author did it like that for a reason and maybe now is no more needed... You can try :wink:

Reply:
I tried it. It seems that it doesn't work.

Reply:
There's a comment about it here (2 lines down):
```
// The prototype is used to override the 'add' function in order to execute the
// following code before the Odoo native override (trick)
```

Reply:
@tarteo well, I saw it but I was wondering if it was still the case :stuck_out_tongue_winking_eye:

Reply:
@simahawk Oh ok haha ;)